src/InMemoryRepository.php

```diff
@@ -261,6 +261,26 @@ private function removeResource(Resource $resource)
     }
 
     /**
+     * {@inheritdoc}
+     */
+    protected function resolveWithLinks($path)
+    {
+        foreach ($this->resources as $resource) {
```

Comment:
I'd maintain a separate list `$links` that keeps a reference of all `LinkResources`. `$resources` can be quite large.